examples/dmd/generic_csv.cpp

```diff
@@ -155,190 +159,234 @@ int main(int argc, char *argv[])
+            indicator_init.push_back(tval);
+
+            csv_db->getDoubleArray(string(data_dir) + "/" + par_dir + "/" + snap_list[snap_list.size()-1] + "/tval.csv", &tval, 1);
+            indicator_last.push_back(tval);
+        }
+
+        CAROM_VERIFY(windowOverlapSamples < windowNumSamples);
+        if (windowNumSamples < infty && indicator_val.size() == 0)
         {
-            cout << "Loading samples for " << par_dir << " to train DMD." << endl;
+            double indicator_min = *min_element(indicator_init.begin(), indicator_init.end());
+            double indicator_max = *max_element(indicator_last.begin(), indicator_last.end());
+            numWindows = ceil((indicator_max - indicator_min) / (dt_est * windowNumSamples));
+            for (int window = 0; window < numWindows; ++window)
+            {
+                indicator_val.push_back(indicator_min + dt_est * windowNumSamples * window);
+            }
+            if (myid == 0)
+            {
+                cout << "Created new indicator range partition." << endl;
+                csv_db->putDoubleVector(string(outputPath) + "/indicator_val.csv", indicator_val, numWindows);
+            }
         }
-        vector<string> snap_list;
-        csv_db->getStringVector(string(list_dir) + "/" + par_dir + ".csv", snap_list, false);
+    }
 
-        int curr_window = 0; // assuming first snapshot falls into first window
-        if (ddt > 0.0)
+    vector<CAROM::DMD*> dmd;
+    dmd.assign(numWindows, nullptr);
+    if (myid == 0)
+    {
+        if (numWindows > 1)
         {
-            dmd[curr_window] = new CAROM::AdaptiveDMD(dim, ddt, "G", "LS", dmd_epsilon);
+            cout << "Using time windowing DMD with " << numWindows << " windows." << endl;
         }
         else
         {
-            dmd[curr_window] = new CAROM::DMD(dim, dtc);
+            cout << "Using serial DMD." << endl;
         }
+    }
 
-        double tval = 0.0;
-        if (indicator_val.size() == 0)
+    for (int window = 0; window < numWindows; ++window)
+    {
+        if (train)
+        {
+            if (ddt > 0.0)
+            {
+                dmd[window] = new CAROM::AdaptiveDMD(dim, ddt, "G", "LS", dmd_epsilon);
+            }
+            else
+            {
+                dmd[window] = new CAROM::DMD(dim, dtc);
+            }
+        }
+        else
         {
-            indicator_val.push_back(tval);
+            if (myid == 0)
+            {
+                cout << "Loading DMD model #" << window << "." << endl;
+            }
+            dmd[window] = new CAROM::DMD(outputPath + "/window" + to_string(window)); 
         }
+    }
 
-        int overlap_count = 0;
-        for (int idx_snap = 0; idx_snap < num_train_snap[idx_dataset]; ++idx_snap)
+    StopWatch dmd_training_timer, dmd_preprocess_timer, dmd_prediction_timer;
+    double* sample = new double[dim];
+    CAROM::Vector* result = new CAROM::Vector(dim, true);
```

Comment:
It looks like `result` is assigned to a pointer to a `Vector` returned by a function. Should it be allocated here?

Reply:
Thanks for the suggestion, Dylan. It is now allocated when assigned by the function.